api/app/Policies/PoolPolicy.php

```diff
@@ -75,40 +76,48 @@ public function viewAnyPublished(?User $user)
      */
     public function create(User $user, $request)
     {
-        if (array_key_exists('team_id', $request)) {
-            $team_id = $request['team_id'];
 
-            // Get the team to check against
-            $team = Team::find($team_id);
+        $teamId = isset($request['team_id']) ? $request['team_id'] : null;
+        $communityId = isset($request['community_id']) ? $request['community_id'] : null;
 
-            // Confirm the user can create pools for the team
-            if (! is_null($team)) {
-                if ($user->isAbleTo('create-team-draftPool', $team)) {
-                    return true;
-                }
-            } else {
-                return Response::deny('Cannot find a team matching team_id.');
-            }
-        } else {
-            Response::deny('Pool must be associated with a team when it is created.');
+        if (is_null($teamId) || is_null($communityId)) {
+            return false;
+        }
+
+        if ($user->isAbleTo('create-any-pool')) {
+            return true; // return early, permission does not exist at the moment
         }
 
-        return Response::deny('Cannot create a pool for that team.');
+        $team = Team::findOrFail($teamId);
+        $community = Community::findOrFail($communityId)->load('team');
```

Comment:
It doesn't make much difference in this case, but it's normally better to eager load the extra with the original request, rather than load it it later.
```suggestion
        $community = Community::with('team')->findOrFail($communityId);
```
https://stackoverflow.com/a/37783791